### PR TITLE
fix: Use splat syntax to avoid errors during destroy with an empty state

### PR DESCRIPTION
**Pull request by barryib**
# PR o'clock

## Description

Resolves https://github.com/terraform-aws-modules/terraform-aws-eks/issues/568

Supersedes:
- https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1040
- https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1020

### Checklist

- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by barryib** [comment]
@@Carles-Figuerola  @ageekymonk Can you please test this and let me know if it solves your issue.

File: local.tf [hunk 1]
```diff
@@ -146,8 +146,8 @@ locals {
 
   kubeconfig = var.create_eks ? templatefile("${path.module}/templates/kubeconfig.tpl", {
     kubeconfig_name                   = local.kubeconfig_name
-    endpoint                          = aws_eks_cluster.this[0].endpoint
-    cluster_auth_base64               = aws_eks_cluster.this[0].certificate_authority[0].data
+    endpoint                          = coalescelist(aws_eks_cluster.this[*].endpoint, [""])[0]
+    cluster_auth_base64               = coalescelist(aws_eks_cluster.this[*].certificate_authority[0].data, [""])[0]
     aws_authenticator_command         = var.kubeconfig_aws_authenticator_command
     aws_authenticator_command_args    = length(var.kubeconfig_aws_authenticator_command_args) > 0 ? var.kubeconfig_aws_authenticator_command_args : ["token", "-i", aws_eks_cluster.this[0].name]
```
Comment (Carles-Figuerola) [inline local.tf:152]:
```suggestion
    aws_authenticator_command_args    = length(var.kubeconfig_aws_authenticator_command_args) > 0 ? var.kubeconfig_aws_authenticator_command_args : ["token", "-i", coalescelist(aws_eks_cluster.this[*].name, [""])[0]]
```

This one is also needed

**Comment by Carles-Figuerola** [comment]
@barryib thanks for consolidating, added a suggestion on a missed line

**Comment by barryib** [comment]
> @barryib thanks for consolidating, added a suggestion on a missed line

Thanks.

**Comment by nick4fake** [comment]
@Carles-Figuerola thanks a lot for the merge!

Could you please create a new release?

**Comment by Carles-Figuerola** [comment]
@nick4fake, sorry, I'm not the one who merged or have write permissions, this is probably on @barryib's terrytory

**Pull request merged**